src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/SnsDomainEventListener.kt

```diff
@@ -0,0 +1,63 @@
+package uk.gov.justice.digital.hmpps.approvedpremisesapi.integration
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import org.junit.jupiter.api.Assertions.fail
+import org.slf4j.LoggerFactory
+import org.springframework.jms.annotation.JmsListener
+import org.springframework.stereotype.Service
+import org.springframework.test.context.event.annotation.BeforeTestMethod
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.model.domainevent.SnsEvent
+
+@Service
+class SnsDomainEventListener(private val objectMapper: ObjectMapper) {
+  private val log = LoggerFactory.getLogger(this::class.java)
+  private val messages = mutableListOf<SnsEvent>()
+
+  @JmsListener(destination = "domaineventsqueue", containerFactory = "hmppsQueueContainerFactoryProxy")
+  fun processMessage(rawMessage: String?) {
+    val (message) = objectMapper.readValue(rawMessage, Message::class.java)
+    val event = objectMapper.readValue(message, SnsEvent::class.java)
+
+    log.info("Received Domain Event: $event")
+    synchronized(messages) {
+      messages.add(event)
+    }
+  }
```

Comment:
👍 some powerful Spring juju, thank you. When you come to finalise this PR it would be great if you could add a reference to some background materials (e.g. introductory blog post on `JmsListener` ) to assist new comers.

Reply:
good idea, done